java/CMakeLists.txt

```diff
@@ -74,6 +74,20 @@ if(BUILD_TESTING)
                                                   GTest::gtest_main)
 endif()
 
+# If the user hasn't set ARROW_JAVA_JNI_ARCH_DIR, derive the normalized
+# operating system from the host processor.
+if("${ARROW_JAVA_JNI_ARCH_DIR}" STREQUAL "")
+  if("${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "aarch64")
```

Comment:
Sorry.
`CMAKE_SYSTEM_PROCESSOR` https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html is suitable for this.
It's the processor for build target.